### PR TITLE
Fix subscriptions via REST when using params

### PR DESCRIPTION
**Pull request by frankosterfeld**
Unify subscription topic handling and fix it for REST (#328, #331)
    
Do not lose params when subscribing to a topic via REST.
    
Bump cpp-httplib to newest version, as it fixes an encoding error with query parameters encoded in query parameters (SubscriptionContext) when used with redirects.
    
To unify subscription handling (different client/servers were making slightly different assumptions), the following is now implemented:
    
 - Service names must always start with "/".
 - The topic is of the form "/some/service?param1=..&param2", where the service name is mandatory in the topic, and the params are optional.
- When serialized to a ZMQ topic, i.e. the string-based PUB/SUB topic matching, the parameters are sorted, to make the subscription and notification topics match exactly (required by ZMQ, as ZMQ subscription topics are string-, not URI-based)
- Valid REST URIs are e.g. "http://localhost:8080/service?a&b" or http://localhost:8080/service?a&b", i.e. the service is a mandatory part of the URI. mds:// and mdp:// URIs are analogous.
- On the ZMQ message level, the "serviceName" in above examples would be "/service" and the "topic" would be "/service?a&b" (here param order is not enforced).

**UPDATE**: Rebased/merged #330 into this one:

Make sure that long-poll request handlers do not block forever when no corresponding event is received. Otherwise the clients will send one request after another once their request times out client-side, until the worker threads are exhausted and the server stops responding.

It would be great if we could also detect the client connection be dropped using Keep-Alive, but cpp-httplib doesn't seem to have API for that.

### CODE REVIEW

**Review by RalphSteinhagen — CHANGES_REQUESTED** [review]
Thanks for the changes. I think we should meet and reiterate on the mapping between URI and MDP protocol. 

The URI's 'path' component should not be broken up on the '/' but map one-to-one to the MDP's 'service name' and the URI's 'query paramters' to the MDP's '[request, reply] topic'. 

I.e. a REST URI '[...]/DeviceName/Acquisition?filterA=...,filterB=...' maps to the MDP:
  * service name = '/DeviceName/Acquisition'
  * request topic = 'filterA=...,filterB=...'
  
Otherwise we mix two different routing concepts. For details see the [extended MDP](https://github.com/fair-acc/opencmw-cpp/blob/main/docs/Majordomo_protocol_comparison.pdf) and -- for reference only --the [original MDP](https://rfc.zeromq.org/spec/18/) definition (N.B. we added, stream-lined and made the in/out commands symmetric compared to the original MDP and more closely following the [URI](https://datatracker.ietf.org/doc/html/rfc3986) definition).

File: concepts/majordomo/FilterSubscription_example.cpp [hunk 1]
```diff
@@ -138,7 +138,7 @@ int main() {
         std::cerr << "Could not bind to broker address" << std::endl;
         return 1;
     }
-    AcquisitionWorker<"/DeviceName/Acquisition"> acquisitionWorker(broker);
```
[the remark this answers is not in the document]
Comment (frankosterfeld) [reply concepts/majordomo/FilterSubscription_example.cpp:141]: Actually, there's another case where mdp/mds and http URIs have the form service/topic?params with non-empty topic, and that's GET/SET to obtain certain objects:

With the opendigitizer dashboards service, the client first requests "/dashboards" to get the list of available dashboards, and then "/dashboards/dashboard1", "/dashboards/dashboard2" ... to get the data for the individual dashboards. Should the "dashboard1", "dashboard2" part be moved to a query param? Like:

"/dashboards" -> returns list of dashboards
"/dashboards?dashboard=dashboard1&what=header" -> returns header of dashboard1

etc.

[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply concepts/majordomo/FilterSubscription_example.cpp:141]: > serviceName: "device1/Acquisition" (no leading slash)

N.B. we discussed/decided earlier that we enforce a leading '/' slash for all services. For the URI this is not strictly enforced by the RFC standard but in 98% of the cases it used as such. The remaining ambiguity would have been whether the `/` is the trailing one and belonging to the authority or leading slash of the 'path' component.

[the remark this answers is not in the document]
Comment (frankosterfeld) [reply concepts/majordomo/FilterSubscription_example.cpp:141]: The issue here is how to parse e.g. `http://localhost:8080/dashboards/dashboard1` (same for mds/mdp://) - the RestBackend can't know if service: `dashboards/dashboard` is meant (if service names can have "/" inside), or service `dashboards`, with topic `dashboard1`.

**Review by RalphSteinhagen — APPROVED** [review]

**Review by ivan-cukic — APPROVED** [review]

**Pull request merged**